yarn-project/simulator/src/public/transitional_adaptors.ts

```diff
@@ -91,149 +70,23 @@ export function createPublicExecutionContext(avmContext: AvmContext, calldata: F
-    write => new ContractStorageUpdateRequest(write.slot, write.value, write.counter.toNumber(), write.storageAddress),
-  );
-  // We need to write the storage updates to the DB, because that's what the ACVM expects.
-  // Assumes the updates are in the right order.
-  for (const write of newWorldState.storageWrites) {
-    await executionContext.stateDb.storageWrite(write.storageAddress, write.slot, write.value);
-  }
-
-  const newNoteHashes = newWorldState.newNoteHashes.map(
-    noteHash => new SideEffect(noteHash.noteHash, noteHash.counter),
-  );
-  const nullifierReadRequests: ReadRequest[] = newWorldState.nullifierChecks
-    .filter(nullifierCheck => nullifierCheck.exists)
-    .map(nullifierCheck => new ReadRequest(nullifierCheck.nullifier, nullifierCheck.counter.toNumber()));
-  const nullifierNonExistentReadRequests: ReadRequest[] = newWorldState.nullifierChecks
-    .filter(nullifierCheck => !nullifierCheck.exists)
-    .map(nullifierCheck => new ReadRequest(nullifierCheck.nullifier, nullifierCheck.counter.toNumber()));
-  const newNullifiers: SideEffectLinkedToNoteHash[] = newWorldState.newNullifiers.map(
-    tracedNullifier =>
-      new SideEffectLinkedToNoteHash(
-        /*value=*/ tracedNullifier.nullifier,
-        /*noteHash=*/ Fr.ZERO, // NEEDED?
-        tracedNullifier.counter,
-      ),
-  );
-  const unencryptedLogs: UnencryptedFunctionL2Logs = new UnencryptedFunctionL2Logs(
-    newWorldState.newLogs.map(log => new UnencryptedL2Log(log.contractAddress, log.selector, log.data)),
-  );
-  const unencryptedLogsHashes = newWorldState.newLogsHashes.map(
-    logHash => new SideEffect(logHash.logHash, logHash.counter),
-  );
-  const newL2ToL1Messages = newWorldState.newL1Messages.map(m => new L2ToL1Message(m.recipient, m.content));
-
-  const returnValues = result.output;
-
-  // TODO: Support nested executions.
-  const nestedExecutions: PublicExecutionResult[] = [];
-  // TODO keep track of side effect counters
-  const startSideEffectCounter = Fr.ZERO;
-  const endSideEffectCounter = Fr.ZERO;
-
-  return {
-    execution,
-    nullifierReadRequests,
-    nullifierNonExistentReadRequests,
-    newNoteHashes,
-    newL2ToL1Messages,
-    startSideEffectCounter,
-    endSideEffectCounter,
-    newNullifiers,
-    contractStorageReads,
-    contractStorageUpdateRequests,
-    returnValues,
-    nestedExecutions,
-    unencryptedLogsHashes,
-    unencryptedLogs,
-    reverted: result.reverted,
-    revertReason: result.revertReason ? createSimulationError(result.revertReason) : undefined,
-    gasLeft: endMachineState.gasLeft,
-  };
 }
 
 export function convertPublicExecutionResult(res: PublicExecutionResult): AvmContractCallResults {
   return new AvmContractCallResults(res.reverted, res.returnValues, res.revertReason);
 }
 
 export function updateAvmContextFromPublicExecutionResult(ctx: AvmContext, result: PublicExecutionResult): void {
```

Comment:
Don't you need to process the nested calls here as well? I'm thinking of the case when AVM calls ACVM and then ACVM has many ACVM nested calls.